tests/testthat/test-guides.R

```diff
@@ -16,6 +16,25 @@ test_that("Colorbar respects show.legend in layer", {
   p <- ggplot(df, aes(x = x, y = y, color = x)) +
     geom_point(size = 20, shape = 21, show.legend = TRUE)
   expect_true("guide-box" %in% ggplotGrob(p)$layout$name)
+
+
+  df <- data.frame(x = 1:3, y = 20:22)
+  p <- ggplot(df, aes(x = x, y = y, color = x, shape = factor(y))) +
+    geom_point(size = 20)
+  expect_length({
+    g <- ggplotGrob(p)
+    g$grobs[[which(g$layout$name == "guide-box")]]
```

Comment:
I think you should pull this out into a helper function. Something like:

```R
n_legends <- function(g) {
  sum(g$layout$name == "guide-box")
}
```

Then you can use `expect_equal()` consistently.